include/machine.h

```diff
@@ -55,5 +64,23 @@ static inline p_region_t CONST pptr_to_paddr_reg(const region_t pptr_reg)
     };
 }
 
+/* Returns the physical region of the kernel image boot part, ie part that is
+ * no longer needed once booting is finished. */
+static inline p_region_t get_p_reg_kernel_img_boot(void)
+{
+    return (p_region_t) {
+        .start = kpptr_to_paddr((const void *)KERNEL_ELF_BASE),
+        .end   = kpptr_to_paddr(ki_boot_end)
+    };
+}
+
+/* Returns the physical region of the kernel image. */
+static inline p_region_t get_p_reg_kernel_img(void)
+{
+    return (p_region_t) {
+        .start = kpptr_to_paddr((const void *)KERNEL_ELF_BASE),
+        .end   = kpptr_to_paddr(ki_end)
+    };
```

Comment:
It seems odd to try and translate pptr's to paddrs, and then have 4 out of the 5 callers translate back to pptr again. pptr is the natural region type as used by everything else in the kernel.